go/vt/vttablet/tabletmanager/rpc_backup.go

```diff
@@ -89,9 +92,17 @@ func (tm *TabletManager) Backup(ctx context.Context, logger logutil.Logger, req
 		if err := tm.changeTypeLocked(ctx, topodatapb.TabletType_BACKUP, DBActionNone, SemiSyncActionUnset); err != nil {
 			return err
 		}
+
+		// adding defer to original value in case of any failures
+		defer func() {
+			bgCtx := context.Background()
+			// Change our type back to the original value.
+			// Original type could be primary so pass in a real value for PrimaryTermStartTime
+			if err := tm.changeTypeLocked(bgCtx, originalType, DBActionNone, SemiSyncActionNone); err != nil {
+				l.Errorf("Fail to change tablet type from %v to %v : %v", topodatapb.TabletType_BACKUP, originalType, err)
```

Comment:
TabletType_BACKUP is an integer (enum index). Instead you probably want to do:
```
l.Errorf("Fail to change tablet type from %s to %s : %v", topodatapb.TabletType_name[int32(topodatapb.TabletType_BACKUP)], topodatapb.TabletType_name[int32(originalType)], err)
```

The numbers will not mean much to a user.

Reply:
When I wrote this code, I tested it and it was not printing the integer instead it was printing the type

./vtctld.Rameezs-MacBook-Pro.rameezsajwani.log.ERROR.20230331-070706.59103:5:E0331 07:07:06.185372   59103 server.go:481] E0331 14:07:06.185178 rpc_backup.go:95] Fail to change tablet type from BACKUP to REPLICA : <nil>

let me know if you still think we should change it ..

Reply:
OK! Might be some protobuf magic then. 🙂 If it works then fine either way -- up to you.